src/components/analyticVisualization/SelectProgram.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState } from 'react'
+import i18n from '@dhis2/d2-i18n'
+import PropTypes from '@dhis2/prop-types'
+import { SelectField } from '../field'
+import { useReadProgramQuery } from '../../pages/Analytics/Program/ProgramVisualizationQueries'
+
+export const SelectProgram = ({ settings, onChange }) => {
+    const { programList, loading } = useReadProgramQuery()
+    const [options, setOptions] = useState([])
+
+    useEffect(() => {
+        if (programList) {
+            setOptions(programList)
+        }
+    }, [programList])
```

Comment:
I think this should work too?

```suggestion
    const { programList, loading } = useReadProgramQuery()
    const options = programList || []
```